internal/command/registry/args.go

```diff
@@ -0,0 +1,203 @@
+	}
+	return machines[index], nil
+}
+
+// argsGetMachineByID returns an app's machine using the `machine` argument.
+func argsGetMachineByID(ctx context.Context, app *fly.AppCompact) (*fly.Machine, error) {
+	flapsClient, err := flapsutil.NewClientWithOptions(ctx, flaps.NewClientOpts{
+		AppCompact: app,
+		AppName:    app.Name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create flaps client for app %s: %w", app.Name, err)
+	}
+
+	machineID := flag.GetString(ctx, "machine")
+	machine, err := flapsClient.Get(ctx, machineID)
+	if err != nil {
+		return nil, err
+	}
+
+	return machine, nil
+}
+
+// argsGetImgPath returns an image path from the command line or from a
+// selected app machine, using `image`, `select`, and `machine`.
+func argsGetImgPath(ctx context.Context, app *fly.AppCompact) (string, error) {
+	if flag.IsSpecified(ctx, "image") {
+		if flag.IsSpecified(ctx, "machine") || flag.IsSpecified(ctx, "select") {
+			return "", fmt.Errorf("image option cannot be used with machien and select options")
+		}
+		return flag.GetString(ctx, "image"), nil
+	}
+
+	machine, err := argsGetMachine(ctx, app)
+	if err != nil {
+		return "", err
+	}
+
+	return imageRefPath(&machine.ImageRef), nil
+}
+
+// argsGetImages returns a list of images in ImgInfo format from
+// command line args or the environment, using `org`, `app`, `running`.
+func argsGetImages(ctx context.Context) ([]ImgInfo, error) {
+	if appName := flag.GetApp(ctx); appName != "" {
+		return argsGetAppImages(ctx, appName)
+	} else if orgName := flag.GetOrg(ctx); orgName != "" {
+		return argsGetOrgImages(ctx, orgName)
+	} else if appName := appconfig.NameFromContext(ctx); appName != "" {
+		return argsGetAppImages(ctx, appName)
+	}
+	return nil, fmt.Errorf("No org or application specified")
+}
+
+// argsGetOrgImages returns a list of images for an org in ImgInfo format
+// from `running`.
+func argsGetOrgImages(ctx context.Context, orgName string) ([]ImgInfo, error) {
+	client := flyutil.ClientFromContext(ctx)
+	org, err := client.GetOrganizationBySlug(ctx, orgName)
+	if err != nil {
+		return nil, err
+	}
+
+	apps, err := client.GetAppsForOrganization(ctx, org.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	var allImgs []ImgInfo
+	for n := range apps {
+		app := &apps[n]
+		imgs, err := argsGetAppImages(ctx, app.Name)
+		if err != nil {
+			return nil, fmt.Errorf("could not fetch images for %q app: %w", app.Name, err)
+		}
+		allImgs = append(allImgs, imgs...)
+	}
+	return allImgs, nil
+
+}
+
+// argsGetAppImages returns a list of images for an app in ImgInfo format
+// from `running`.
+func argsGetAppImages(ctx context.Context, appName string) ([]ImgInfo, error) {
+	apiClient := flyutil.ClientFromContext(ctx)
+	app, err := apiClient.GetAppCompact(ctx, appName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get app %q: %w", appName, err)
+	}
+
+	flapsClient, err := flapsutil.NewClientWithOptions(ctx, flaps.NewClientOpts{
+		AppCompact: app,
```

Comment:
I get both the AppID and OrgID from this query and need to use both later in the code. I'll have to work through this one a little to see if I can loosen that to do less queries (maybe just once for the org info).

Reply:
Cool. Don't let me derail you too much. We can always circle back and make it run faster if it feels slow for big orgs.

Reply:
I updated flapsutil to use OrgSlug instead of AppCompact when passed. I got rid of the unnecessary AppID in image info. I was able to avoid doing AppCompact lookup for every app when scanning all apps in an org.

Reply:
Those of us with slow internet thank you

Reply:
thats me :)